ts/components/MessageComponent.tsx

```diff
@@ -8,22 +8,24 @@ import mapPropsToStyleNames from "native-base/src/Utils/mapPropsToStyleNames";
 import I18n from "../i18n";
 import { ServicesState } from "../store/reducers/entities/services";
 
-export type OwnProps = {
+export type OwnProps = Readonly<{
   sender: string;
   subject: string;
   key: string;
   date: string;
-  services: Readonly<ServicesState>;
-};
+  services: ServicesState;
+}>;
 
 export type Props = OwnProps;
 
 /**
  * Implements a component that show a message in the MessagesScreen List
  */
 class MessageComponent extends React.Component<Props> {
-  public formatDate(date: string): string {
+  // Convert the distance from now to date in : H.mm, yesterday, D/MM/YY and DD/MM
```

Comment:
then you should probably not use `distanceInWordsToNow`, instead calculate the difference from now in days and decide what to do based on the number of days - also as suggested by @gunzip , extract this logic into a function in the `ts/utils` folder since this is not some logic specific to this component